src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1053,7 +1053,20 @@ GrpcLb::GrpcLb(const grpc_lb_addresses* addresses,
   GPR_ASSERT(server_uri != nullptr);
   grpc_uri* uri = grpc_uri_parse(server_uri, true);
   GPR_ASSERT(uri->path[0] != '\0');
-  server_name_ = gpr_strdup(uri->path[0] == '/' ? uri->path + 1 : uri->path);
```

Comment:
Thanks for doing this!

One thing I'm not sure of: does this grpclb code need to work when the server uri's scheme is not DNS?

Reply:
I think this single line should always work whatever the scheme?

What do you think would be the possible failure case?